File: Nunit.Test/Examples/System/PropertyExamples.cs
using System;
using NUnit.Framework;

namespace Examples.Examples.System {
	[TestFixture()]
	class PropertyExamples {

		/// <summary>
		/// When only accessed to get or set there is no need for implementing get/set
		/// </summary>
		public string AutoProperty { get; set; }
		
		private string _property;
		/// <summary>
		/// Property with implementation in get or set
		/// </summary>
		public string Property {
			get {
				if (string.IsNullOrEmpty(_property)) {
					_property = Guid.NewGuid().ToString();
				}
				return _property;
			}
			set {
				_property = value;
			}
		}


		/// <summary>
		/// To have different access modifiers, simply declare them on get or set
		/// </summary>
		public string AutoPropertyWithRestricedAccessModifier { get; protected set; }

		/// <summary>
		/// Property with implementation in get or set and different access modifiers
		/// </summary>
		public string PropertyWithRestricedAccessModifier {
			get
			{
				if (string.IsNullOrEmpty(_property)) {
					_property = Guid.NewGuid().ToString();
				}
				return _property;
			}
			protected set {
				_property = value;
			}
		}

		/// <summary>
		/// Starting with C# 6, read-only properties can implement the get accessor as an expression-bodied member
		/// </summary>
		public string ReadOnlyProperty => Guid.NewGuid().ToString();
	}
}

File: NUnit.vb.Test/Examples/System/PropertyExamples.vb
Namespace Examples.System
	Public Class PropertyExamples

		''' <summary>
		''' When only accessed to get or set there is no need for implementing get/set
		''' </summary>
		Public Property AutoProperty As String

		Private _property As String

		''' <summary>
		''' Property with implementation in get or set
		''' </summary>
		Public Property [Property] As String
			Get
				If String.IsNullOrEmpty(_property) Then
					_property = Guid.NewGuid().ToString()
				End If

				Return _property
			End Get
			Set(ByVal value As String)
				_property = value
			End Set
		End Property

		''' <summary>
		''' To have different access modifiers, simply declare them on get or set
		''' </summary>
		Public Property PropertyWithRestricedAccessModifier As String
			Get
				If String.IsNullOrEmpty(_property) Then
					_property = Guid.NewGuid().ToString()
				End If

				Return _property
			End Get
			Protected Set(ByVal value As String)
				_property = value
			End Set
		End Property

	End Class
End Namespace